app/models/smart_proxy.rb

```diff
@@ -115,11 +115,12 @@ def associate_features
 
     begin
       reply = ProxyAPI::Features.new(:url => url).features
-      if reply.is_a?(Array) && reply.any?
```

Comment:
I think the Array check here on `reply` is important - it's moved below to `valid_features.is_a?(Array)`, but you know that's an array as you're constructing it with .map. Check the `reply` is an array before trying to use map on it.